src/parse-mockdb.js

```diff
@@ -4,52 +4,175 @@ const Parse = require('parse-shim');
 const _ = require('lodash');
 
 const DEFAULT_LIMIT = 100;
-const HARD_LIMIT = 1000;
-const MAX_SKIP = 10000;
 const QUOTE_REGEXP = /(\\Q|\\E)/g;
 
 const CONFIG = {
-  DEBUG: process.env.DEBUG_DB
+  DEBUG: process.env.DEBUG_DB,
+};
+
+let db = {};
+let hooks = {};
+const masks = {};
+
+let indirect = null;
+let outOfBandResults = null;
+
+let defaultController = null;
+let mocked = false;
+
+function debugPrint(prefix, object) {
+  if (CONFIG.DEBUG) {
+    console.log(['[', ']'].join(prefix), JSON.stringify(object, null, 4));
+  }
 }
 
-const HANDLERS = {
-  GET: handleGetRequest,
-  POST: handlePostRequest,
-  PUT: handlePutRequest,
-  DELETE: handleDeleteRequest,
+function isOp(object) {
+  return object && typeof object === 'object' && '__op' in object;
 }
 
-var db = {};
-var hooks = {};
-var masks = {};
+function isPointer(object) {
+  return object && object.__type === 'Pointer';
+}
 
-var indirect = null;
-var outOfBandResults = null;
-var default_controller = null;
-var mocked = false;
+function isDate(object) {
+  return object && object.__type === 'Date';
+}
 
 /**
- * Mocks a Parse API server, by intercepting requests and storing/querying data locally
- * in an in-memory DB.
+ * Deserialize an encoded query parameter if necessary
  */
-function mockDB() {
-  if (!mocked) {
-    default_controller = Parse.CoreManager.getRESTController();
-    mocked = true;
-    Parse.CoreManager.setRESTController(MockRESTController);
+function deserializeQueryParam(param) {
+  if (!!param && (typeof param === 'object')) {
+    if (param.__type === 'Date') {
+      return new Date(param.iso);
+    }
   }
+  return param;
 }
 
 /**
- * Restores the original RESTController.
+ * Evaluates whether 2 objects are the same, independent of their representation
+ * (e.g. Pointer, Object)
  */
-function unMockDB() {
-  if (mocked) {
-    Parse.CoreManager.setRESTController(default_controller);
-    mocked = false;
+function objectsAreEqual(obj1, obj2) {
+  // scalar values (including null/undefined)
+  // eslint-disable-next-line eqeqeq
```

Comment:
in this case, we want  undefined and null to be treated the same, so == is appropriate cause the if (this && that) to get the four permutations would be lame?